### PR TITLE
test: deflake tests

### LINKED ISSUE #110: Tests are randomly failing
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.0.0

### Plugin version

2.0.1

### Node.js version

18, 20, 21

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

*

### Description

When running `npm test` sometimes tests fail with a timeout, but it is not repeatable on demand.  The error has happened on Linux, Windows, and MacOS.

### Steps to Reproduce

`npm run test`

https://github.com/fastify/fastify-swagger-ui/actions/runs/6776721797/job/18418830040
https://github.com/fastify/fastify-swagger-ui/actions/runs/7216288278/job/19662243321

```
# Subtest: test/integration.test.js
    # Subtest: fastify will response swagger csp
        1..1
        ok 1 - should be equivalent
    ok 1 - fastify will response swagger csp # time=591.307ms
    
    1..1
    # time=597.533ms
not ok 4 - test/integration.test.js # time=597.533ms
  ---
  env: {}
  file: test/integration.test.js
  timeout: 30000
  command: /opt/hostedtoolcache/node/18.19.0/x64/bin/node
  args:
    - test/integration.test.js
  stdio:
    - 0
    - pipe
    - 2
  cwd: /home/runner/work/fastify-swagger-ui/fastify-swagger-ui
  exitCode: null
  signal: SIGTERM
  ...
```

```
not ok 4 - test/integration.test.js # time=675.262ms
  ---
  env: {}
  file: test/integration.test.js
  timeout: 30000
  command: C:\hostedtoolcache\windows\node\21.1.0\x64\node.exe
  args:
    - test/integration.test.js
  stdio:
    - 0
    - pipe
    - 2
  cwd: D:\a\fastify-swagger-ui\fastify-swagger-ui
  exitCode: null
  signal: SIGTERM
  ...
```

```
​ FAIL ​ test/route.test.js 89 OK 344.544ms
  command: /Users/user/.nvm/versions/node/v20.8.0/bin/node
  args:
    - test/route.test.js
  exitCode: null
  signal: SIGTERM

​ FAIL ​ TAP
 ✖ test/route.test.js

  test: TAP
  env: {}
  file: test/route.test.js
  timeout: 30000
  command: /Users/user/.nvm/versions/node/v20.8.0/bin/node
  args:
    - test/route.test.js
  stdio:
    - 0
    - pipe
    - 2
  cwd: /Users/user/git/open/fastify-swagger-ui
  exitCode: null
  signal: SIGTERM
```

### Expected Behavior

_No response_

### PR DESCRIPTION
**Pull request by Uzlopak**
Closes #110 
I guess we should have awaited some of the .register() calls and make it more robust.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Uzlopak** [comment]
I ran the CI twice now and no tests failed.

**Review by Fdawgs — APPROVED** [review]

File: test/integration.test.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ test('fastify will response swagger csp', async (t) => {
 
   await fastify.register(fastifySwagger)
   await fastify.register(fastifySwaggerUi)
-  fastify.register(fastifyHelmet, instance => {
+  await fastify.register(fastifyHelmet, instance => {
```
Comment (mcollina) [inline test/integration.test.js:21]: This is likely the only needed await. Can you verify?

Comment (Uzlopak) [reply test/integration.test.js:21]: All of them are needed.

Comment (mcollina) [reply test/integration.test.js:21]: How? I don't think so.

File: test/route.test.js [hunk 2]
```diff
@@ -192,7 +192,7 @@ test('/v1/foobar should redirect to ./foobar/static/index.html - in plugin', asy
   t.plan(3)
   const fastify = Fastify()
 
-  fastify.register(async function (fastify, options) {
+  await fastify.register(async function (fastify, options) {
```
Comment (mcollina) [inline test/route.test.js:195]: This is not needed, beucase .inject() would register all plugins.

File: test/route.test.js [hunk 3]
```diff
@@ -217,7 +217,7 @@ test('/v1/foobar/ should redirect to ./static/index.html - in plugin', async (t)
...
```
Comment (mcollina) [inline test/route.test.js:220]: This is not needed, inject() will register all plugins.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: test/route.test.js (hunk 3 above)
Comment (Uzlopak) [reply test/route.test.js:220]: Didnt know that .inject will finalize the encapsulation.

```suggestion
  fastify.register(async function (fastify, options) {
```

File: test/route.test.js (hunk 2 above)
Comment (Uzlopak) [reply test/route.test.js:195]:
```suggestion
  fastify.register(async function (fastify, options) {
```

**Comment by Uzlopak** [comment]
@mcollina 
PTAL

;)

File: test/route.test.js (hunk 3 above)
Comment (Fdawgs) [reply test/route.test.js]: Aw fudge, I didn't know this either. It's there in the documentation as well!

And here I am awaiting `ready` in my own repo tests prior to injecting.

File: index.js [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
 
+const fs = require('node:fs')
+const path = require('node:path')
 const fp = require('fastify-plugin')
 const { readFile } = require('node:fs/promises')
-const path = require('node:path')
 
 async function fastifySwaggerUi (fastify, opts) {
-  fastify.decorate('swaggerCSP', require('./static/csp.json'))
+  const csp = JSON.parse(await fs.promises.readFile(path.resolve(__dirname, './static/csp.json'), 'utf8'))
```
Comment (mcollina) [inline index.js:9]: The reason why we use require is to be bundler friendly if possible.

**Comment by mcollina** [comment]
disabling code coverage would make the test pass, there is a bug in node

**Comment by Uzlopak** [comment]
@mcollina 

Well, i dont know anymore. I tried to improve the code at best to avoid any race conditions or whatever. I quite dont get why it fails. But if you are ok, i would like to merge.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**